LeaderboardBackend.Test/Leaderboards.cs

```diff
@@ -348,4 +349,190 @@ public async Task GetLeaderboards()
+        {
+            Jwt = jwt,
+        });
+
+        await act.Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.Forbidden);
+    }
+
+    [Test]
+    public async Task RestoreLeaderboard_NotFound()
+    {
+        Func<Task<LeaderboardViewModel>> act = async () => await _apiClient.Put<LeaderboardViewModel>($"/leaderboard/{1e10}/restore", new()
+        {
+            Jwt = _jwt
+        });
+
+        await act.Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.NotFound);
+    }
+
+    [Test]
+    public async Task RestoreLeaderboard_NotFound_WasNeverDeleted()
+    {
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        Leaderboard board = new()
+        {
+            Name = "Hyper Mario World Not Deleted",
+            Slug = "hyper-mario-world-non-deleted",
+        };
+
+        context.Leaderboards.Add(board);
+        await context.SaveChangesAsync();
+        board.Id.Should().NotBe(default);
+
+        try
+        {
+            await _apiClient.Put<LeaderboardViewModel>(
+                $"/leaderboard/{board.Id}/restore",
+                new()
+                {
+                    Jwt = _jwt,
+                }
+            );
+        }
+        catch (RequestFailureException e)
+        {
+            e.Response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+            string? model = await e.Response.Content.ReadAsStringAsync();
+            model!.Should().MatchRegex("Not Deleted");
+        }
+    }
+
+    [Test]
+    public async Task RestoreLeaderboard_Conflict()
+    {
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        Leaderboard deleted = new()
+        {
+            Name = "Conflicted Mario World",
+            Slug = "conflicted-mario-world",
+            DeletedAt = _clock.GetCurrentInstant()
+        };
+
+        Leaderboard reclaimed = new()
+        {
+            Name = "Reclaimed Mario World",
+            Slug = "conflicted-mario-world",
+        };
+
+        context.Leaderboards.Add(deleted);
+        context.Leaderboards.Add(reclaimed);
+        await context.SaveChangesAsync();
+
+        try
+        {
+            await _apiClient.Put<LeaderboardViewModel>(
+                $"/leaderboard/{deleted.Id}/restore",
+                new()
+                {
+                    Jwt = _jwt,
+                }
+            );
+        }
+        catch (RequestFailureException e)
+        {
+            e.Response.StatusCode.Should().Be(HttpStatusCode.Conflict);
+            LeaderboardViewModel? model = await e.Response.Content.ReadFromJsonAsync<LeaderboardViewModel>(TestInitCommonFields.JsonSerializerOptions);
+            model.Should().NotBeNull();
+            model!.Slug.Should().Be("conflicted-mario-world");
+        }
```

Comment:
This test is written incorrectly because in the case that request succeeds, the test doesn't fail (bad). Check the discord for a better way to write this.